### PR TITLE
Modification in Face-Detection/face-detect.py

### LINKED ISSUE #657: Modification in Face-Detection/face-detect.py
**Problem**
The code is not fully optimized and have less feature in it.

**Solution I'd like to add**
So I want to add more feature and optimized python code using open-cv in it
Feature I will be working upon is:

1) Which Detects the Mid part in the face.
2) Detects the eyes and face part , by Putting them in rectangle frame as image.

**Additional context**
I will be highly Obliged if you assign this task to me So please assign this updating part to me so I can update this code and add some feature and value in it.
Thank You :))

### PR DESCRIPTION
**Pull request by anubhav201241**
# Description

The things that I fixed are the following:
1) Firstly I modified the code and add some features to it.
2) I modified the code by adding an eye detection feature and added the haarcascade_eye.xml file.
3) I modified the code by adding the mid detection part in the face 

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [ ] Yes

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update

## Checklist:

- [ ] My code follows the style guidelines(Clean Code) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have created a helpful and easy to understand `Code`
- [ ] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by anubhav201241** [comment]
For the issue no:- #657

**Comment by santushtisharma10** [comment]
@kaustubhgupta @antrikshmisri  pr #667 is also an eye detection program.

**Comment by kaustubhgupta** [comment]
@santushtisharma10 Actually @anubhav201241's approach is improving face detection script with eye detection and PR #667 is dedicated to eye detection. I think we can adopt both of them

**Comment by anubhav201241** [comment]
Yes, but I add the eye detection part and detecting the mid part of the face because to make the code good and easy to understand, I have done many changes to make it understandable and to optimized it, I do because I think it will be better if I provide full-face detection with detecting the eyes, nose, and face which makes it optimized and easy to understand if you don't want eye part I can commit the change and delete that eye_detecs part but I think it will be better if we keep them because it makes the code easier to understand :))

File: Face-Detection/face_detect.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#Face and Eye Detection 
+import cv2
+
+#PATH of the FILE
+CLASSIFIER_PATH = "Facedetection/haarcascade_frontalface_default.xml"
```
Comment (kaustubhgupta) [inline Face-Detection/face_detect.py:5]:
```suggestion
CLASSIFIER_PATH = "./Facedetection/haarcascade_frontalface_default.xml"   
```

File: Face-Detection/face_detect.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+#Face and Eye Detection 
+import cv2
+
+#PATH of the FILE
+CLASSIFIER_PATH = "Facedetection/haarcascade_frontalface_default.xml"   
+EYEDETECTION_PATH = "Facedetection/haarcascade_eye.xml"
```
Comment (kaustubhgupta) [inline Face-Detection/face_detect.py:6]:
```suggestion
EYEDETECTION_PATH = "./Facedetection/haarcascade_eye.xml"
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Rename the file face_detect.py to face_detectv2.py. Other changes:

**Review by kaustubhgupta — COMMENTED** [review]
Wait, you don't need to make changes to the current file. Create a folder called v2 and name the script as face_detectv2.py and put the code there. This is because, in another PR #681, he will also do the same thing as v3. Create a readme for this version and shift the model files XML to v2 folder

File: Face-Detection/face_detect.py (hunk 2 above)
Comment (anubhav201241) [reply Face-Detection/face_detect.py]: Sir I have made the changes, please check it.

File: Face-Detection/v2/face_detectv2.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+#Face and Eye Detection 
+import cv2
+
+#PATH of the FILE
+CLASSIFIER_PATH = "./Facedetection/haarcascade_frontalface_default.xml"
```
Comment (kaustubhgupta) [inline Face-Detection/v2/face_detectv2.py:5]:
```suggestion
CLASSIFIER_PATH = "./Facedetection/v2/haarcascade_frontalface_default.xml"   
```

File: Face-Detection/v2/face_detectv2.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
+#Face and Eye Detection 
+import cv2
+
+#PATH of the FILE
+CLASSIFIER_PATH = "./Facedetection/haarcascade_frontalface_default.xml"   
+EYEDETECTION_PATH = "./Facedetection/haarcascade_eye.xml"
```
Comment (kaustubhgupta) [inline Face-Detection/v2/face_detectv2.py:6]:
```suggestion
EYEDETECTION_PATH = "./Facedetection/v2/haarcascade_eye.xml"
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Add haarcascade_frontalface_default.xml in v2 folder. Don't shift it from Face-Detection folder. Upload new file in v2 folder

**Comment by anubhav201241** [comment]
Sir, I had updated the code and folder please look into it

**Review by santushtisharma10 — CHANGES_REQUESTED** [review]
@anubhav201241 add a readme file in v2 folder

File: Face-Detection/v2/face_detectv2.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    # Displaying video feed with detected faces
+    cv2.imshow('Frame', frame)
+
+    # Reading for keyboard interrupts
+    key = cv2.waitKey(1)
+    if (key == 27):
+        break
+
+cam.release()
+cv2.destroyAllWindows
```
Comment (antrikshmisri) [inline Face-Detection/v2/face_detectv2.py:53]: You didn't call the function
```suggestion
cv2.destroyAllWindows()
```

**Review by antrikshmisri — CHANGES_REQUESTED** [review]

File: Face-Detection/v2/face_detectv2.py (hunk 5 above)
Comment (anubhav201241) [reply Face-Detection/v2/face_detectv2.py]: Updated it, Sir, Please check, and thank you for letting me know

**Review by kaustubhgupta — APPROVED** [review]

**Review by santushtisharma10 — APPROVED** [review]
@anubhav201241 💯

**Review by antrikshmisri — APPROVED** [review]
LGTM 👍🏻

**Review by avinashkranjan — APPROVED** [review]

**Pull request merged**